locustfiles/all.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Tests the Support API.
+    """
+
+    local_port = "9443"
+    domain = MilMoveDomain.PRIME
+    is_api = True
+
+    parser = support_api
+
+    wait_time = between(0.25, 9)
+    tasks = {SupportTasks: 1}
+
+
+class ServicesCounselorUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the MilMove Office app with the Services Counselor role.
+    """
+
+    local_protocol = "http"
+    local_port = "8080"
+    domain = MilMoveDomain.OFFICE
+
+    # This attribute is used for generating fake requests when hitting the GHC API:
+    parser = ghc_api
+
+    wait_time = between(0.25, 9)
+    tasks = {ServicesCounselorTasks: 1}
+
+
+class TOOUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the MilMove Office app with the TOO role.
+    """
+
+    local_protocol = "http"
+    local_port = "8080"
+    domain = MilMoveDomain.OFFICE
+
+    # This attribute is used for generating fake requests when hitting the GHC API:
+    parser = ghc_api
+
+    wait_time = between(0.25, 9)
+    tasks = {TOOTasks: 1}
+
+
+class MilMoveUser(MilMoveHostMixin, HttpUser):
+    """
+    Tests the MilMove app.
+    """
+
+    local_protocol = "http"
+    local_port = "8080"
+    domain = MilMoveDomain.MILMOVE
+
+    wait_time = between(1, 9)
+    tasks = {MilMoveTasks: 1}
+
+
+@events.test_stop.add_listener
+def on_test_stop(**kwargs):
+    """
+    Clean up steps to run when the load test stops. Removes any cert files that may have been created during the setup.
+    """
+    clean_milmove_host_users(locust_env=kwargs["environment"])
+
+
+file_path = os.path.dirname(os.path.abspath(__file__))
+
+extend = Blueprint(
+    "extend",
+    "extend_web_ui",
+    static_folder=f"{file_path}/static/",
+    static_url_path="/extend/static/",
+    template_folder=f"{file_path}/templates/",
+)
+
+
+@events.init.add_listener
+def on_locust_init(environment, **kw):
+
+    if environment.web_ui:
+        original_user_classes = {user_class.__name__: user_class for user_class in environment.user_classes}
+
+        @extend.route("/extend")
+        def get_user_classes():
+            current_user_classes = [user_class.__name__ for user_class in environment.user_classes]
+            environment.web_ui.update_template_args()
+            render_ctx = environment.web_ui.template_args.copy()
+            render_ctx["original_user_classes"] = original_user_classes.keys()
+            render_ctx["current_user_classes"] = current_user_classes
+            return render_template("get_user_classes.html", **render_ctx)
+
+        @extend.route("/set_user_classes", methods=["POST"])
+        def set_user_classes():
+            assert request.method == "POST"
+
+            class_options = request.form.getlist("user_classes_option")
+            new_classes = [original_user_classes[n] for n in class_options]
+            setattr(environment, "user_classes", new_classes)
+            environment._remove_user_classes_with_weight_zero()
```

Comment:
This errors out for me, because it's an internal method I'm guessing? Since you're setting the user_classes above to what was selected we wouldn't need to set the unselected classes to a weight of zero I'm assuming.

Reply:
I *think* this might be in a different version of locust. It doesn't error out for me, so hopefully the pipenv changes will make it work for you, too